designer/client/src/containers/Notifications.tsx

```diff
@@ -3,56 +3,75 @@ import { default as ReactNotifications } from "react-notification-system-redux";
-    return ReactNotifications.show(
-        {
-            autoDismiss: autoDismiss,
-            uid: backendNotification.id,
-            children: [<Notification icon={icon} message={backendNotification.message} key={uuid4()} />],
-            onRemove: () => {
-                dispatch(markBackendNotificationRead(backendNotification.id));
-            },
-        },
-        backendNotification.type,
-    );
-}
+const prepareNotification =
+    ({ id, message, type }: BackendNotification): ThunkAction =>
+    (dispatch) => {
+        if (!type) {
+            //if we don't display notification, we assume that it's already processed
+            return dispatch(markBackendNotificationRead(id));
+        }
+        return dispatch(
+            ReactNotifications.show(
+                {
+                    autoDismiss: type == "error" ? 0 : 10,
+                    uid: id,
+                    children: <Notification icon={type == "error" ? <TipsError /> : <TipsSuccess />} message={message} />,
+                    onRemove: () => dispatch(markBackendNotificationRead(id)),
+                },
+                type,
+            ),
+        );
+    };
 
-function handleRefresh(beNotification: BackendNotification, currentScenarioName, dispatch: Dispatch<any>) {
-    const scenarioName = beNotification.scenarioName;
-    if (scenarioName && scenarioName == currentScenarioName) {
-        beNotification.toRefresh.forEach((data) => {
+const handleRefresh =
+    ({ scenarioName, toRefresh }: BackendNotification, currentScenarioName: string): ThunkAction =>
+    (dispatch) => {
+        if (!scenarioName || scenarioName !== currentScenarioName) {
+            return;
+        }
+        toRefresh.forEach((data) => {
             switch (data) {
                 case "versions":
-                    dispatch(loadProcessVersions(beNotification.scenarioName));
-                    break;
+                    return dispatch(loadProcessVersions(scenarioName));
                 case "activity":
-                    dispatch(displayProcessActivity(beNotification.scenarioName));
-                    break;
+                    return dispatch(displayProcessActivity(scenarioName));
                 case "state":
-                    dispatch(loadProcessState(beNotification.scenarioName));
+                    return dispatch(loadProcessState(scenarioName));
             }
         });
-    }
-}
+    };
+
+const prepareNotifications =
+    (notifications: BackendNotification[], scenarioName: string): ThunkAction =>
+    (dispatch, getState) => {
+        const state = getState();
+        const { processedNotificationIds } = getBackendNotifications(state);
+        const reactNotifications = getNotifications(state);
+
+        const onlyUnreadPredicate = ({ id }: BackendNotification) =>
+            !processedNotificationIds.includes(id) && !reactNotifications.map(({ uid }) => uid).includes(id);
```

Comment:
We can write this condition `!reactNotifications.map(({ uid }) => uid).includes(id);` like

```
!reactNotifications.some(({ uid }) => uid === id);
```